local_units/management/commands/import-local-units-csv.py

```diff
@@ -19,57 +20,86 @@ def add_arguments(self, parser):
+                is_public = row['PUBLICVIEW'].lower() == 'yes'
+                validated = row['VALIDATED'].lower() == 'yes'
+                local_branch_name = row['NAME_LOC']
+                english_branch_name = row['NAME_EN']
+                postcode = row['POSTCODE'].strip()[:10]
+                address_loc = row['ADDRESS_LOC']
+                address_en = row['ADDRESS_EN']
+                city_loc = row['CITY_LOC']
+                city_en = row['CITY_EN']
+                focal_person_loc = row['FOCAL_PERSON_LOC']
+                focal_person_en = row['FOCAL_PERSON_EN']
+                phone = row['TELEPHONE'].strip()[:30]
+                email = row['EMAIL']
+                link = row['WEBSITE']
+                source_en = row['SOURCE_EN']
+                source_loc = row['SOURCE_LOC']
+                location = Point(float(row['LONGITUDE']), float(row['LATITUDE']))
                 if row['DATEOFUPDATE']:
-                    unit.date_of_data = date_parser.parse(row['DATEOFUPDATE']).replace(tzinfo=pytz.utc)
-                unit.save()
-                name = unit.local_branch_name if unit.local_branch_name else unit.english_branch_name
-                city = unit.city_loc if unit.city_loc else unit.city_en
-                if name:
-                    print(f'{i} | {name} saved')
-                elif city:
-                    print(f'{i} | ** {city} city location saved')
-                else:
-                    print(f'{i} | *** entity with ID saved')
+                    date_of_data = datetime.strptime(row['DATEOFUPDATE'], '%d-%b-%y').strftime("%Y-%m-%d")
+                local_units_to_create.append(
+                    LocalUnit(
+                        country=country,
+                        type=type,
+                        level=level,
+                        subtype=subtype,
+                        is_public=is_public,
+                        validated=validated,
+                        local_branch_name=local_branch_name,
+                        english_branch_name=english_branch_name,
+                        postcode=postcode,
+                        address_en=address_en,
+                        address_loc=address_loc,
+                        city_en=city_en,
+                        city_loc=city_loc,
+                        focal_person_en=focal_person_en,
+                        focal_person_loc=focal_person_loc,
+                        phone=phone,
+                        email=email,
+                        link=link,
+                        source_en=source_en,
+                        source_loc=source_loc,
+                        location=location,
+                        date_of_data=date_of_data
+                    )
+                )
+        bulk_mgr = BulkCreateManager(chunk_size=1000)
+        for local_unit in local_units_to_create:
+            bulk_mgr.add(local_unit)
+        bulk_mgr.done()
+        print('Bulk create summary:', bulk_mgr.summary())
```

Comment:
Let's do this something like this
```python
      ....
        with open(filename) as csvfile:
            reader = csv.DictReader(csvfile)
            bulk_mgr = BulkCreateManager(chunk_size=1000)
            
            for i, row in enumerate(reader):
                .........
                bulk_mgr.add(local_unit)
           bulk_mgr.done()
           print('Bulk create summary:', bulk_mgr.summary())
```
Basically replace local_units_to_create with bulk_mgr